### PR TITLE
this, used in extension functions

### PR DESCRIPTION
**Pull request by deeper-x**

### CODE REVIEW

File: _examples/ThisExtensionFunc.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+title: this in extension functions
+---
+    
+<div class="sample" markdown="1">
+
+Normally, when you create a class, by default it inherits from *Any*, the Kotlin default supertype which declares toString(), hashCode() and equals().
```
Comment (hhariri) [inline _examples/ThisExtensionFunc.md]: @B1ggDave please review constructs.

Comment (B1ggDave) [inline _examples/ThisExtensionFunc.md]: looks good, a tiny suggestion:
Normally, when you create a class, it inherits from **Any** (the Kotlin default supertype). **Any** declares toString(), hashCode(), and equals().

File: _examples/ThisExtensionFunc.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+In other languages like Java, if you want to extend a class behaviour, you create a *derived* class like this:
```
Comment (B1ggDave) [inline _examples/ThisExtensionFunc.md]: behavior 
if we are keeping Ktor as American English?

File: _examples/ThisExtensionFunc.md (hunk 1 above)
Comment (deeper-x) [reply _examples/ThisExtensionFunc.md]: Good point.

File: _examples/ThisExtensionFunc.md (hunk 2 above)
Comment (B1ggDave) [reply _examples/ThisExtensionFunc.md]: you would need to create a *derived* class like this:

File: _examples/ThisExtensionFunc.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+```
+
+1. Class that allows to be inheritable. Note: Any is inherted by default
+2. Initializer block, with some code
+3. Derived class definition
+4. Instance creation, with inherited behaviour  
+  
+But in Kotlin, with **extension functions** you have another way to extend class functionalities: just declare a new *fun* on the base class, no need to define another (derived) class and generate its instance.
```
Comment (B1ggDave) [inline _examples/ThisExtensionFunc.md]: But in Kotlin, with **extension functions** you have another way to extend class functionalities: simply declare a new *fun* on the base class, there is no need to define another (derived) class and generate its instance.

File: _examples/ThisExtensionFunc.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+In this example, you can see how it's easy to declare an extension and, with keyword **this**, access the object, called *receiver*:
```
Comment (B1ggDave) [inline _examples/ThisExtensionFunc.md]: In this example, you can see how easy it is to declare an extension and, with keyword **this**, access the object, called *receiver*:

File: _examples/ThisExtensionFunc.md (hunk 2 above)
Comment (deeper-x) [reply _examples/ThisExtensionFunc.md]: :%s/behaviour/behavior/g 

I learned my lesson! :D

**Comment by AlexanderPrendota** [comment]
@deeper-x Hi! Could you resolve conflicts and put your example to new [`examples`](https://github.com/kotlinbyexample/kotlinbyexample.github.io/tree/master/examples) folder.

See information [here](https://github.com/kotlinbyexample/kotlinbyexample.github.io#how-to-contribute)

**Comment by deeper-x** [comment]
Conflicts on _data/examples.yml solved.

File: _examples/ThisExtensionFunc.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+fun main(args: Array<String>){
+    val person = Developer()           //4 
+}
+
+```
+
+1. Class that allows to be inheritable. Note: Any is inherted by default
+2. Initializer block, with some code
+3. Derived class definition
+4. Instance creation, with inherited behaviour
```
[the remark this answers is not in the document]
Comment (deeper-x) [reply _examples/ThisExtensionFunc.md]: :%s/behaviour/behavior/g

I learned my lesson! :D

**Comment by AlexanderPrendota** [comment]
Hi! Thank you for contributing. Please create the new PR and resolve all conflicts. Make sure you use new templates for examples.

Thanks! :)

**Pull request closed without merging**